src/allencell_ml_segmenter/core/file_input_widget.py

```diff
@@ -256,49 +267,60 @@ def _process_checked_signal(self, row: int, state: Qt.CheckState) -> None:
             elif state == Qt.CheckState.Unchecked:
                 # could have unselected the img we got channels from originally, so need to re-extract
                 # as long as there are still some images selected
-                if len(selected_indices) == 0:
-                    self._service.stop_channel_extraction()  # stop so combobox doesn't reset after thread is finished
+                if (
+                    len(selected_indices) == 0
+                    and self._include_channel_selection
+                ):
+                    if self._service:
+                        self._service.stop_channel_extraction()  # stop so combobox doesn't reset after thread is finished
                     self._reset_channel_combobox()
 
                 self._model.set_selected_paths(
-                    selected_paths, extract_channels=len(selected_paths) > 0
+                    selected_paths,
+                    extract_channels=len(selected_paths) > 0
+                    and self._include_channel_selection,
                 )
 
     def _reset_channel_combobox(self) -> None:
-        self._channel_select_dropdown.clear()
-        self._channel_select_dropdown.setPlaceholderText("")
-        self._channel_select_dropdown.setCurrentIndex(-1)
-        self._channel_select_dropdown.setEnabled(False)
+        if self._include_channel_selection:
```

Comment:
is this condition about this being a prediction vs thresholding use case?

Needing this all over says to me that the abstraction need some work.